RewardPool.sol

```diff
@@ -0,0 +1,204 @@
+
+abstract contract IRewardDistributor is Ownable {
+    address rewardDistributor;
+
+    constructor(address _rewardDistributor) public {
+        rewardDistributor = _rewardDistributor;
+    }
+
+    function notifyRewardAmount(uint256 reward) external virtual;
+
+    modifier onlyRewardDistributor() {
+        require(_msgSender() == rewardDistributor, "Caller is not reward distribution");
+        _;
+    }
+
+    function setRewardDistributor(address _rewardDistributor)
+        external
+        onlyOwner
+    {
+        rewardDistributor = _rewardDistributor;
+    }
+}
+
+contract LPTokenWrapper {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    IERC20 public lpToken;
+
+    uint256 private _totalSupply;
+    mapping(address => uint256) private _balances;
+
+    function totalSupply() public view returns (uint256) {
+        return _totalSupply;
+    }
+
+    function balanceOf(address account) public view returns (uint256) {
+        return _balances[account];
+    }
+
+    function stake(uint256 amount) public virtual {
+        _totalSupply = _totalSupply.add(amount);
+        _balances[msg.sender] = _balances[msg.sender].add(amount);
+        lpToken.safeTransferFrom(msg.sender, address(this), amount);
+    }
+
+    function withdraw(uint256 amount) public virtual {
+        require(_balances[msg.sender] >= amount, "withdraw: insufficient balance");
+        _totalSupply = _totalSupply.sub(amount);
+        _balances[msg.sender] = _balances[msg.sender].sub(amount);
+        lpToken.safeTransfer(msg.sender, amount);
+    }
+}
+
+
+/// @title A reward pool that does not mint
+/// @dev The rewards should be first transferred to this pool, then get "notified" by calling `notifyRewardAmount`.
+///      Only the reward distributor can notify.
+contract RewardPool is LPTokenWrapper, IRewardDistributor, ReentrancyGuard {
+
+    using Address for address;
+
+    IERC20 public rewardToken;
+    uint256 public duration;
+
+    uint256 public periodFinish = 0;
+    uint256 public rewardRate = 0;
+    uint256 public lastUpdateTime;
+    uint256 public rewardPerTokenStored;
+    mapping(address => uint256) public userRewardPerTokenPaid;
+    mapping(address => uint256) public rewards;
+
+    event RewardAdded(uint256 reward);
+    event Staked(address indexed user, uint256 amount);
+    event Withdrawn(address indexed user, uint256 amount);
+    event RewardPaid(address indexed user, uint256 reward);
+    event RewardDenied(address indexed user, uint256 reward);
+
+    modifier updateReward(address account) {
+        rewardPerTokenStored = rewardPerToken();
+        lastUpdateTime = lastTimeRewardApplicable();
+        if (account != address(0)) {
+            rewards[account] = earned(account);
+            userRewardPerTokenPaid[account] = rewardPerTokenStored;
+        }
+        _;
+    }
+
+    constructor(
+        address _rewardToken,
+        address _lpToken,
```

Comment:
Since we are going to have DRC as lp maybe move it to the constants and rename to dracula?